darwin/torch/dataset.py

```diff
@@ -63,194 +68,7 @@ def get_dataset(
-                f"Could not find any {SUPPORTED_IMAGE_EXTENSIONS} file",
-                f" in {images_dir}"
-            )
-
-        assert len(self.images_path) == len(self.annotations_path)
-
-    def extend(self, dataset, extend_classes: bool = False):
-        """Extends the current dataset with another one
-
-        Parameters
-        ----------
-        dataset : Dataset
-            Dataset to merge
-        extend_classes : bool
-            Extend the current set of classes by merging with the passed dataset ones
-
-        Returns
-        -------
-        Dataset
-            self
-        """
-        if self.annotation_type != dataset.annotation_type:
-            raise ValueError("Annotation type of both datasets should match")
-        if self.classes != dataset.classes and not extend_classes:
-            raise ValueError(
-                f"Operation dataset_a + dataset_b could not be computed: classes "
-                f"should match. Use flag extend_classes=True to combine both lists "
-                f"of classes."
-            )
-        self.classes = list(set(self.classes).union(set(dataset.classes)))
-
-        self.original_images_path = self.images_path
-        self.images_path += dataset.images_path
-        self.original_annotations_path = self.annotations_path
-        self.annotations_path += dataset.annotations_path
-        return self
-
-    def get_img_info(self, index: int):
-        with self.annotations_path[index].open() as f:
-            data = json.load(f)["image"]
-            return data
-
-    def get_height_and_width(self, index: int):
-        data = self.get_img_info(index)
-        return data["height"], data["width"]
-
-    def _map_annotation(self, index: int):
-        """
-        Load an annotation and filter out the extra classes according to what
-        specified in `self.classes`
-
-        Parameters
-        ----------
-        index : int
-            Index of the annotation to read
-
-        Returns
-        -------
-        dict
-        A new dictionary containing the index and the filtered annotation
-
-        Notes
-        -----
-        The return value is a dict with the following fields:
-            image_id: int
-                The index of the image in the split
-            original_filename: str
-                The path to the image on the file system
-            annotations : str
-                The original raw annotation
-        """
-        with self.annotations_path[index].open() as f:
-            annotation = json.load(f)["annotations"]
-        # Filter out unused classes
-        if self.classes is not None:
-            annotation = [a for a in annotation if a["name"] in self.classes]
-        return {
-            "image_id": index,
-            "original_filename": self.images_path[index],
-            "annotations": annotation,
-        }
+class DarwinDataset(LocalDataset):
 
     def measure_mean_std(self, multi_threaded: bool = True):
```

Comment:
Looks like `DarwinDataset` is just a `LocalDataset` with `measure_mean_std` in addition to it.
Why don't we use `DarwinDataset` right away instead? In other words, is there a reason to keep `LocalDataset` separate from this `dataset.py`?

Reply:
The main idea for `LocalDataset` is to be a parser of the json files with some basic functionality that can be used in the different modules in `darwin-py`: right now it's only used by the `torch` module, but eventually could be used by the `tf` module, `dl` module, or any other function that needs to access to the json annotations. That's why `LocalDataset` is in `darwin.dataset` and not in `darwin.torch`.

The reason I created `DarwinDataset` in `darwin.torch` is because `measure_mean_std` requires the `pillow` module (or `cv2`) to load images, so I didn't want to add that dependency outside of `darwin.torch`. If you think it's okay adding this dependency to `darwin-py` then I can move those functions to `LocalDataset` and get rid of `DarwinDataset`.

Reply:
No, I think yours is a valid point, let's keep it this way then!